selftest.py doctest failures under Python 3

Need to adjust expected outputs for `__pycache__` directories.

Python 3.2.2
nose revision 551ffe92

```
======================================================================
FAIL: Doctest: imported_tests.rst
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/unix/aksarkar/.local/lib/python3.2/doctest.py", line 2118, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for imported_tests.rst
  File "/home/unix/aksarkar/.local/src/nose/build/tests/functional_tests/doc_tests/test_issue145/imported_tests.rst", line 0

----------------------------------------------------------------------
File "/home/unix/aksarkar/.local/src/nose/build/tests/functional_tests/doc_tests/test_issue145/imported_tests.rst", line 14, in imported_tests.rst
Failed example:
    print(ls_tree(support)) # doctest: +REPORT_NDIFF
Differences (ndiff with -expected +actual):
      |-- package1
      |   |-- __init__.py
    - |   `-- test_module.py
    ?     ^
    + |   |-- test_module.py
    ?     ^
    + |   `-- __pycache__
    + |       |-- __init__.cpython-32.pyc
    + |       `-- test_module.cpython-32.pyc
      |-- package2c
      |   |-- __init__.py
    - |   `-- test_module.py
    ?     ^
    + |   |-- test_module.py
    ?     ^
    + |   `-- __pycache__
    + |       |-- __init__.cpython-32.pyc
    + |       `-- test_module.cpython-32.pyc
      `-- package2f
          |-- __init__.py
    -     `-- test_module.py
    ?     ^
    +     |-- test_module.py
    ?     ^
    +     `-- __pycache__
    +         |-- __init__.cpython-32.pyc
    +         `-- test_module.cpython-32.pyc


======================================================================
FAIL: Doctest: selector_plugin.rst
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/unix/aksarkar/.local/lib/python3.2/doctest.py", line 2118, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for selector_plugin.rst
  File "/home/unix/aksarkar/.local/src/nose/build/tests/functional_tests/doc_tests/test_selector_plugin/selector_plugin.rst", line 0

----------------------------------------------------------------------
File "/home/unix/aksarkar/.local/src/nose/build/tests/functional_tests/doc_tests/test_selector_plugin/selector_plugin.rst", line 22, in selector_plugin.rst
Failed example:
    print(ls_tree(support))
Expected:
    |-- mymodule.py
    |-- mypackage
    |   |-- __init__.py
    |   |-- strings.py
    |   `-- math
    |       |-- __init__.py
    |       `-- basic.py
    `-- tests
        |-- testlib.py
        |-- math
        |   `-- basic.py
        |-- mymodule
        |   `-- my_function.py
        `-- strings
            `-- cat.py
Got:
    |-- mymodule.py
    |-- __pycache__
    |   `-- mymodule.cpython-32.pyc
    |-- mypackage
    |   |-- __init__.py
    |   |-- strings.py
    |   |-- __pycache__
    |   |   |-- __init__.cpython-32.pyc
    |   |   `-- strings.cpython-32.pyc
    |   `-- math
    |       |-- __init__.py
    |       |-- basic.py
    |       `-- __pycache__
    |           |-- __init__.cpython-32.pyc
    |           `-- basic.cpython-32.pyc
    `-- tests
        |-- testlib.py
        |-- __pycache__
        |   `-- testlib.cpython-32.pyc
        |-- math
        |   |-- basic.py
        |   `-- __pycache__
        |       `-- basic.cpython-32.pyc
        |-- mymodule
        |   |-- my_function.py
        |   `-- __pycache__
        |       `-- my_function.cpython-32.pyc
        `-- strings
            |-- cat.py
            `-- __pycache__
                `-- cat.cpython-32.pyc


----------------------------------------------------------------------
```

--- a/nose/util.py
+++ b/nose/util.py
@@ -20,7 +20,7 @@ log = logging.getLogger('nose')
 
 ident_re = re.compile(r'^[A-Za-z_][A-Za-z0-9_.]*$')
 class_types = (ClassType, TypeType)
-skip_pattern = r"(?:\.svn)|(?:[^.]+\.py[co])|(?:.*~)|(?:.*\$py\.class)"
+skip_pattern = r"(?:\.svn)|(?:[^.]+\.py[co])|(?:.*~)|(?:.*\$py\.class)|(?:__pycache__)"
 
 try:
     set()

